packages/grafana-ui/src/components/Box/Box.tsx

```diff
@@ -0,0 +1,54 @@
+import { css } from '@emotion/css';
+import React, { useCallback } from 'react';
+
+import { GrafanaTheme2, ThemeSpacingTokens } from '@grafana/data';
+
+import { useStyles2 } from '../../themes';
+
+type ExtendSpacing = {
+  vertical?: ThemeSpacingTokens;
+  horizontal?: ThemeSpacingTokens;
+  top?: ThemeSpacingTokens;
+  bottom?: ThemeSpacingTokens;
+  left?: ThemeSpacingTokens;
+  right?: ThemeSpacingTokens;
+};
+type Spacing = ThemeSpacingTokens | ExtendSpacing;
+
+interface BoxProps {
+  margin?: Spacing;
+  padding?: Spacing;
+  backgroundColor?: keyof GrafanaTheme2['colors']['background'];
+}
+
+export const Box = ({ children, ...props }: React.PropsWithChildren<BoxProps>) => {
```

Comment:
I think this should accept a `element` prop for changing the html component from day one.

it also should forwardRef

Reply:
oh, and i think `display: block | inline` to start with.

Reply:
I struggle to see when this component should be used, can't think of many scenarios where I would want a block element box like this.

Reply:
Box is/will be essentially a "guided" way to write CSS without writing CSS using the grafana design tokens. It's the lowest level component in the design system, and theoretically all design system components can be made from Box.

It's not there yet, which is why it's a draft PR.

We've been discussing using Alert as an example - that should be able to be build entirely from Boxes.

Reply:
I can't see how that could work, Alert needs flex layout elements with gap / flex options.  really struggle to see any use it. Other than occasional "marginBottom" on an element

Reply:
This is a draft PR. It is not complete!

Reply:
Hypothetically, in pseudo-code, it could be used something like this

```ts
<Box borderRadius="small" borderColor="danger" padding="medium" background="dangerMuted">
  <Stack direction="horizontal" gap="gutter">
    <Icon name="hazard" size="medium" />
    
    <Stack direction="vertical" gap="small">
      <Text weight="bold">{title}</Text>
      {description && <Text>{description}</Text>}
    </Stack>
    
    {onAction && <Button style="ghost" onClick={onAction}>Close</Button>}
  </Stack>
</Box>
```

Reply:
@joshhunt ah, now I understand. That makes some sense :+1: